src/common/snippets/include/snippets/lowered/linear_ir.hpp

```diff
@@ -139,6 +136,74 @@ class LinearIR {
     const std::shared_ptr<ShapeInferSnippetsNode>& get_shape_infer_instance() const {return m_shape_infer; }
     VectorDims get_master_shape() const;
 
+    /* ------ Helpers for work with LinearIR ----- */
```

Comment:
Regarding all the helpers: in general, I think that using ov::Node in LIR helpers is not a good practice because the Node can't fully describe the LIR data flow, and the validity of ov::model connections is not guaranteed at this point.
Although I agree that it is a tempting compromise for the `insert_node` methods (at least a user can provide connectors manually), I think it's worth to consider changing `replace_node` signatures to work with expressions only.
Please see more detailed comments in the corresponding methods.

Reply:
Good!